### PR TITLE
feature/58_SORTING_FUNCTIONALITY

### PR DESCRIPTION
**Pull request by Symphonie234**
### Changes Made:

- Created a new method in the **`Listing`** **Model** for sorting listings by the most frequently associated tags.
- Added code in the **`HomeController`** which sorts listings based on a 'sort' query parameter, prioritizing options such as the latest, top-rated by likes, or defaulting to sorting by the most frequently used tag.
- Adjusted the code in the **`home/index.blade.php`** for the "Latest, Relevant, and Top" sorting to work.
- Added a new **JavaScript** file named **`sorting.js`** for handling sorting of listings.

### CODE REVIEW

**Comment by Symphonie234** [comment]
### Additional Changes Made:

- Changed the pagination for listings in the **`HomeController`** **index** function from 5 to 10.

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -28,8 +23,22 @@ public function index()
...
             ->get();
 
-        // Fetch all listings from the database
-        $listings = Listing::orderBy('created_at', 'desc')->paginate(5);
+        // Start with a base query for listings
+        $listingsQuery = Listing::query();
+
+        // Sort the listings based on the 'sort' query parameter
+        $sort = $request->query('sort');
+        if ($sort === 'latest') {
```
Comment (ChronoDevs) [inline app/Http/Controllers/HomeController.php]: make an enum and your queries please transfer to the model

File: app/Http/Controllers/HomeController.php [hunk 2]
```diff
@@ -28,8 +23,22 @@ public function index()
             ->take(10) // Limit to the top 10 tags
```
Comment (ChronoDevs) [inline app/Http/Controllers/HomeController.php]: set limit from const file or enum

File: app/Http/Controllers/HomeController.php [hunk 3]
```diff
@@ -9,12 +9,7 @@
 
 class HomeController extends Controller
 {
-    /**
-     * Show the application dashboard.
-     *
-     * @return \Illuminate\Contracts\Support\Renderable
-     */
-    public function index()
+    public function index(Request $request)
```
Comment (ChronoDevs) [inline app/Http/Controllers/HomeController.php:18]: add comment method above

**Comment by Symphonie234** [comment]
### Changes Addressed in the PR No. 1:

- Made a new **Enum** file named **`ListingSort.php`**.
- Modified the **`HomeController`** to shorten it.
- Added two new functions in the **`Listing`** **Model** named **`sortBy`** and **`mostUsedTags`**

**Pull request merged**